bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -172,9 +173,33 @@ public boolean tableExists(String tableName) throws IOException {
     return response;
   }
 
+  // Workaround for #2016, BigtableTableAdminClient#getTable throws TableNotFoundException(due to
+  // two step details fetch) and an intermediate exception(FAILED_PRECONDITION).
+  //
+  // This method is only being called when user sends `null` or `""` string to listTable(Pattern)
+  // which in being used by deleteTables(), enableTables(), disableTables(),
+  // getTableDescriptorByTableNames().
+  private HTableDescriptor[] getTableDescriptorsIgnoreFailure(TableName[] tableNames) {
+    List<HTableDescriptor> descriptors = new ArrayList<>();
+    for (TableName tableName : tableNames) {
+      try {
+        descriptors.add(getTableDescriptor(tableName));
+      } catch (IOException ex) {
+        // Do we need better error message around it.
+        LOG.warn(
```

Comment:
This is an expected scenario, no need to alarm the user. Please remove the logging